stratum/hal/lib/tdi/tdi_sde_table_entry.cc

```diff
@@ -171,16 +171,24 @@ ::util::Status TdiSdeWrapper::GetTableEntry(
   RET_CHECK(real_table_data);
   const ::tdi::Table* table;
   RETURN_IF_TDI_ERROR(tdi_info_->tableFromIdGet(table_id, &table));
-
   const ::tdi::Device* device = nullptr;
   ::tdi::DevMgr::getInstance().deviceGet(dev_id, &device);
   std::unique_ptr<::tdi::Target> dev_tgt;
   device->createTarget(&dev_tgt);
 
   const auto flags = ::tdi::Flags(0);
-  RETURN_IF_TDI_ERROR(table->entryGet(*real_session->tdi_session_, *dev_tgt,
-                                      flags, *real_table_key->table_key_,
-                                      real_table_data->table_data_.get()));
+  tdi_status_t status = table->entryGet(*real_session->tdi_session_, *dev_tgt,
+                                        flags, *real_table_key->table_key_,
+                                        real_table_data->table_data_.get());
+
+  if (status == TDI_TABLE_NOT_FOUND || status == TDI_OBJECT_NOT_FOUND) {
+    return MAKE_ERROR(::util::error::Code::NOT_FOUND).without_logging()
+           << "No matching table entry";
+  } else if (status != TDI_SUCCESS) {
+    return MAKE_ERROR(::util::error::Code::INTERNAL).without_logging()
```

Comment:
Agree about returning original error instead of mapping to ERR_INTERNAL. Fixed it for this change with your pro-tip :)

Will create a follow up PR to fix other cases in stratum later.

For your 2nd question,
When erroring out, SDE layer logs  the reason of failure and p4_service layers also emit a log similar to `Failed to read forwarding entries from node 1: One or more read operations failed`.  

Current stratum layer also emits a log in tdi_table_manager layer and tdi_sde_table_entry layer. So, we have total 4 logs for 1 error which is not desirable. This PR reduces it to 2 error logs( 1 from SDE from TDI call perspective and other from stratum from P4RT call perspective)

Reply:
We should consider doing this for all TDI calls after checking if SDE is logging for failures. If it is, then upper layer shouldn't as long as it doesn't have any new information